src/code/z_collision_check.c

```diff
@@ -1,13 +1,53 @@
 #include "global.h"
 #include "vt.h"
+#include "overlays/effects/ovl_Effect_Ss_HitMark/z_eff_ss_hitmark.h"
 
-// draw red poly
-void func_8005B280(GraphicsContext* gfxCtx, Vec3f* vA, Vec3f* vB, Vec3f* vC) {
-    func_8005B2AC(gfxCtx, vA, vB, vC, 255, 0, 0);
+typedef s32 (*ColChkResetFunc)(GlobalContext*, Collider*);
+typedef void (*ColChkBloodFunc)(GlobalContext*, Collider*, Vec3f*);
+typedef void (*ColChkApplyFunc)(GlobalContext*, CollisionCheckContext*, Collider*);
+typedef void (*ColChkVsFunc)(GlobalContext*, CollisionCheckContext*, Collider*, Collider*);
+typedef s32 (*ColChkLineFunc)(GlobalContext*, CollisionCheckContext*, Collider*, Vec3f*, Vec3f*);
+
+typedef struct {
+    /* 0 */ u8 blood;
+    /* 1 */ u8 effect;
+} HitInfo; // size = 0x2
+
+typedef enum {
+    /* 0 */ NO_BLOOD,
+    /* 1 */ BLUE_BLOOD,
+    /* 2 */ GREEN_BLOOD,
+    /* 3 */ WATER_BURST,
+    /* 4 */ RED_BLOOD,
+    /* 5 */ RED_BLOOD2
```

Comment:
I suggest using blood as a prefix for this enum, to make things a bit cleaner and more consistent